src/swtpm/swtpm_nvstore_linear_file.c

```diff
@@ -0,0 +1,274 @@
+#include "config.h"
+
+#define _GNU_SOURCE
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/stat.h>
+#include <sys/mman.h>
+#include <sys/ioctl.h>
+#include <linux/fs.h>
+#include <fcntl.h>
+
+#include <libtpms/tpm_error.h>
+
+#include "swtpm.h"
+#include "swtpm_debug.h"
+#include "swtpm_nvstore_linear.h"
+#include "swtpm_nvstore_linear_file.h"
+#include "logging.h"
+#include "tpmstate.h"
+
+/*
+    Provides a linear backend based on memory-mapping a filesystem path.
+    Can be used to access regular files (with automatic resizing) or block
+    devices (with pre-allocated constant size, must be big enough for all state,
+    otherwise writing will fail).
+*/
+struct nvram_linear_store_ops nvram_linear_file_ops = {
+    .open = SWTPM_NVRAM_LinearFile_Open,
+    .flush = SWTPM_NVRAM_LinearFile_Flush,
+    .resize = SWTPM_NVRAM_LinearFile_Resize,
+};
```

Comment:
this should go to the bottom of the file and all functions it uses should become static and the header file can then go.